Extension/src/Debugger/attachToProcess.ts

```diff
@@ -167,6 +177,58 @@ export class RemoteAttachPicker {
         }
     }
 
+    private async getRemoteProcessesExtendedRemote(miDebuggerPath: string, miDebuggerServerAddress: string): Promise<AttachItem[]> {
+        const args: string[] = [`-ex "target extended-remote ${miDebuggerServerAddress}"`, '-ex "info os processes"', '-batch'];
+        let processListOutput: util.ProcessReturnType = await util.spawnChildProcess(miDebuggerPath, args);
+        // The device may not be responsive for a while during the restart after image deploy. Retry 5 times.
+        for (let i: number = 0; i < 5 && !processListOutput.succeeded; i++) {
+            processListOutput = await util.spawnChildProcess(miDebuggerPath, args);
+        }
+
+        if (!processListOutput.succeeded) {
+            throw new Error(localize('failed.to.make.gdb.connection', 'Failed to make GDB connection: "{0}".', processListOutput.output));
+        }
+        const processes: AttachItem[] = this.parseProcessesFromInfoOsProcesses(processListOutput.output);
+        if (!processes || processes.length === 0) {
+            throw new Error(localize('failed.to.parse.processes', 'Failed to parse processes: "{0}".', processListOutput.output));
+        }
+        return processes;
+    }
+
+    /**
+    Format:
+    pid      usr      command     cores
+    1        ?
+    2        ?
+    3                 /usr/bin/sample 0
+    4        root     /usr/bin/gdbserver --multi :6000 0
+
+    Returns aa array AttachItems, each one contains a label of "<user   >command", and a pid.
+    Unfortunately becuase the format of each line is not fixed, and everything except pid is optional, it's hard
+    to get a better label.
+    */
+    private parseProcessesFromInfoOsProcesses(processList: string): AttachItem[] {
+        const lines: string[] = processList?.split('\n');
+        if (!lines?.length) {
+            return [];
+        }
+
+        const processes: AttachItem[] = [];
+        for (const line of lines) {
+            const trimmedLine: string = line.trim();
+            if (!trimmedLine.endsWith('?') && !trimmedLine.includes('/usr/bin/gdbserver')) {
```

Comment:
Will there be an issue if gdbserver is not in /usr/bin/ but in another path?

Reply:
There won't be an issue but gdbserver will show up in the process picker.

That said... People probably will use gdbserver to debug another gdbserver process... So I'll remove this check.